bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingReadOp.java

```diff
@@ -580,18 +590,23 @@ void sendReadTo(int bookieIndex, BookieId to, LedgerEntryRequest entry) throws I
 
     @Override
     public void readEntryComplete(int rc, long ledgerId, final long entryId, final ByteBuf buffer, Object ctx) {
+
         final ReadContext rctx = (ReadContext) ctx;
         final LedgerEntryRequest entry = rctx.entry;
 
         if (rc != BKException.Code.OK) {
-            entry.logErrorAndReattemptRead(rctx.bookieIndex, rctx.to, "Error: " + BKException.getMessage(rc), rc);
+            // if entry complete = true, don't need logErrorAndReattemptRead
+            if (!entry.complete.get()) {
```

Comment:
we should add a test covering this change: calling readEntryComplete with a non-OK error code twice